dubbo-remoting/dubbo-remoting-api/src/main/java/org/apache/dubbo/remoting/transport/AbstractClient.java

```diff
@@ -113,7 +111,12 @@ public AbstractClient(URL url, ChannelHandler handler) throws RemotingException
 
     protected static ChannelHandler wrapChannelHandler(URL url, ChannelHandler handler) {
         url = ExecutorUtil.setThreadName(url, CLIENT_THREAD_POOL_NAME);
-        url = url.addParameterIfAbsent(Constants.THREADPOOL_KEY, Constants.DEFAULT_CLIENT_THREADPOOL);
+        String key = Constants.DEFAULT_KEY + Constants.HIDE_KEY_PREFIX + Constants.THREADPOOL_KEY;
+        String threadpool = url.getParameter(key);
+        if (StringUtils.isNotEmpty(threadpool)) {
+            url = url.removeParameter(key);
+        }
+        url = url.addParameterIfAbsent(Constants.THREADPOOL_KEY, StringUtils.isNotEmpty(threadpool) ? threadpool : Constants.DEFAULT_CLIENT_THREADPOOL);
```

Comment:
Do we need to change the code here?
I think we will have `default.threadpool` key in automatically.

Reply:
Ok, because I found that when ReferenceConfig integrate the configuration of ConsumerConfig, it will add the prefix "default.", so the "threadpool" property of <dubbo:consumer ../> will change to "default.thread" parameter in the URL.
So, I think we have two ways to optimize the issue.
@chickenlj First, take some special process when integrate the "threadpool" property of ConsumerConfig to ReferenceConfig to wipe off the prefix "default."；
Second is just the method that this sumbit uses.
If the second way is not so good, I think we can take the second one.

Reply:
Look at the `getParameter("threadpool")` logic in URL, it will: first check "threadpool" and then check "default.threadpool"

https://github.com/apache/incubator-dubbo/blob/0b6e6f3a375d82aab1a1f17b034e666c63f61103/dubbo-common/src/main/java/org/apache/dubbo/common/URL.java#L421